wallet/src/main/java/com/android/identity_credential/wallet/MainActivity.kt

```diff
@@ -108,50 +125,78 @@ class MainActivity : ComponentActivity() {
 
     private val provisioningViewModel: ProvisioningViewModel by viewModels()
     private val credentialInformationViewModel: CredentialInformationViewModel by viewModels()
+    private lateinit var sharedPreferences: SharedPreferences
 
     private val permissionTracker = PermissionTracker(this, mapOf(
         Manifest.permission.CAMERA to "This application requires camera permission to scan",
         Manifest.permission.NFC to "NFC permission is required to operate"
     ))
 
+    private val inPersonPresentationPermissionTracker: PermissionTracker = if (Build.VERSION.SDK_INT >= 31) {
+        PermissionTracker(this, mapOf(
+            Manifest.permission.BLUETOOTH_ADVERTISE to "This application requires Bluetooth " +
+                    "advertising to send credential data",
+            Manifest.permission.BLUETOOTH_SCAN to "This application requires Bluetooth " +
+                    "scanning to send credential data",
+            Manifest.permission.BLUETOOTH_CONNECT to "This application requires Bluetooth " +
+                    "connection to send credential data"
+        ))
+    } else {
+        PermissionTracker(this, mapOf(
+            Manifest.permission.ACCESS_FINE_LOCATION to "This application requires Bluetooth " +
+                    "to send credential data"
+        ))
+    }
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
         application = getApplication() as WalletApplication
+        sharedPreferences = PreferenceManager.getDefaultSharedPreferences(applicationContext)
 
         permissionTracker.updatePermissions()
+        inPersonPresentationPermissionTracker.updatePermissions()
+        val requiredPermissions: List<String> = if (Build.VERSION.SDK_INT >= 31) {
```

Comment:
Should be more descriptive name than just requiredPermissions (at least if declared on top level rather than locally next to its use). This set is for a particular screen or workflow, right?